File: dbTest.js
import mysql from 'mysql2';

//#region 取得全域變數
import dotenv from 'dotenv';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';

// 取得當前文件的絕對路徑 (__dirname 的替代方案)
const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

// 使用絕對路徑加載 .env 文件
dotenv.config({ path: join(__dirname, '../../.env') });
import { getOrdersUrl } from '../../config.js'; // 從 config.js 中引入 getOrdersUrl函數
//#endregion

// 建立資料庫連線池
const pool = mysql.createPool({
    host: process.env.EC2_PRIVATE_IP,
    user: 'root',
    password: 'mysql',
    database: 'ShoppingSiteDB',
});

const promisePool = pool.promise();

// 測試連接的自執行異步函數
(async () => {
    try {
        const [rows, fields] = await promisePool.query('SELECT 1');
        console.log('Query result:', rows); // 應該回傳 [{ '1': 1 }]
    } catch (error) {
        console.error('Database connection error:', error);
    }
})();

File: db_test.py
# db_test.py
from pymongo import MongoClient
import os
from dotenv import load_dotenv
from config import get_orders_url  # 從 config.py 中引入 get_orders_url 函數

# 載入環境變數
load_dotenv()

# 從環境變數取得 MongoDB 連線資訊
MONGODB_URI = f"mongodb://{os.getenv('EC2_PRIVATE_IP', 'localhost')}:27017"
DB_NAME = 'ShoppingSiteDB'

# 建立 MongoDB 連線
client = MongoClient(MONGODB_URI)
db = client[DB_NAME]

# 測試連接的自執行異步函數
def test_connection():
    try:
        # 執行一個簡單的查詢
        result = db.command('ping')
        print('Query result:', result)  # 應該回傳 { 'ok': 1 }
    except Exception as error:
        print('Database connection error:', error)

# 調用測試函數
if __name__ == '__main__':
    test_connection()
